pylint/config/utils.py

```diff
@@ -0,0 +1,34 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Utils for arguments/options parsing and handling."""
+
+
+from typing import Any, Dict
+
+from pylint.config.argument import _Argument
+
+IMPLEMENTED_OPTDICT_KEYS = {"action", "default", "type", "choices", "help", "metavar"}
+"""This is used to track our progress on accepting all optdict keys."""
+
+
+def _convert_option_to_argument(opt: str, optdict: Dict[str, Any]) -> _Argument:
+    """Convert an optdict to an Argument class instance."""
+    # See if the optdict contains any keys we don't yet implement
+    # pylint: disable-next=fixme
+    # TODO: This should be removed once the migration to argparse is finished
+    for key, value in optdict.items():
+        if key not in IMPLEMENTED_OPTDICT_KEYS:
+            print("Unhandled key found in Argument creation:", key)  # pragma: no cover
+            print("It's value is:", value)  # pragma: no cover
```

Comment:
Can this be seen by end user with custom plugin ? Maybe we could add a link to our issue tracker ?

Reply:
I don't think so (right now). Plugins will need to start using the `argparse` system and as long as we don't tell about them they probably won't.

I hope to make `argparse` the standard for `pylint's` own checkers in `2.14` and then start telling people about them. That would be after we fix all these open issues though.